### PR TITLE
feat: `docker run` option

### LINKED ISSUE #89: `docker run` option
## description
see https://www.reddit.com/r/PostgreSQL/s/Y4spwXqGmG

## proposed solution
example: https://github.com/ajayd-san/gomanagedocker

### PR DESCRIPTION
**Pull request by ftupas**
This pr does the following:
- Adds Dockerfile to build the `rainfrog` binary
- Update readme
- resolves #89 

```shell
❯ docker run -it --rm --name rainfrog \
  -p 5432:5432 \
  -e username="rainfrog" \
  -e password="password" \
  -e hostname="host.docker.internal" \
  -e db_port="5432" \
  -e dbname="rainfrog" rainfrog
```
image

### CODE REVIEW

**Comment by ftupas** [comment]
For now we build it locally but I think docker hub needs to be setup in order to 
- Publish the image to docker hub so it can be pulled directly
- Add cicd for docker

File: Dockerfile
Comment (achristmascarl) [inline Dockerfile:1]: curious, what's the reason for different images for runtime vs building?

Comment (ftupas) [reply Dockerfile:1]: primarily for separation of concerns but also for efficiency and maintainability, bonus is you have a smaller image size

here's a great reference https://docs.docker.com/build/building/multi-stage/

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+WORKDIR /usr/src/app
+
+# Create non-root user
+RUN useradd -m -s /bin/bash rainfrog
+
+# Copy the binary from the builder image
+COPY --chmod=755 --from=builder /app/target/release/rainfrog /usr/local/bin/rainfrog
+
+# Change ownership of the files to the non-root user
+RUN chown -R rainfrog:rainfrog /usr/src/app
```
Comment (achristmascarl) [inline Dockerfile:22]: if the binary is in /usr/local/bin, does this chown do anything?

Comment (ftupas) [reply Dockerfile:22]: good catch, updated it

**Pull request merged**